lib/services/suppressed-violations-manager.js

```diff
@@ -0,0 +1,275 @@
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Requirements
+//-----------------------------------------------------------------------------
+
+const fs = require("node:fs");
+const path = require("node:path");
+const { calculateStatsPerFile } = require("../eslint/eslint-helpers");
+
+//-----------------------------------------------------------------------------
+// Exports
+//-----------------------------------------------------------------------------
+
+/**
+ * Manages the suppressed violations.
+ */
+class SuppressedViolationsManager {
+    suppressionsLocation = "";
+    cwd = "";
+
+    /**
+     * Creates a new instance of SuppressedViolationsManager.
+     * @param {string} suppressionsLocation The location of the suppressions file.
+     * @param {string} cwd The current working directory.
+     */
+    constructor(suppressionsLocation, cwd) {
```

Comment:
Can we make this an options object?
```suggestion
    constructor({ filePath, cwd }) {
```

I think `filePath` is a bit more descriptive and a bit less typing.